### DISCUSSION TITLE
How to get face rotation angles in degrees?

### DISCUSSION THREAD

**Discussion by IgorSamer** [vladmandic/human#313] [Q&A] [answered]
Hi! I noticed that in the default labels the rotation angle values are shown in degrees. But looking into `face.rotation.angles` the result is:

`{ pitch: -0.12624927651314846, yaw: 0.34794775363718944, roll: -0.1220533528063716 }`

What is the correct calc to transform these values into degrees as in labels?

Thanks!

**Comment by IgorSamer** [comment]
Okay, after a few tries I found out that just multiplying the values by 60 to do the conversion :)

**Comment by vladmandic** [comment, marked as answer]
close, but not really :)

values are in radians, so to get degrees its `(angle * 180 / Math.PI)` which is **57.2957795** - so *almost* 60 :)

**Reply by kekoleg** [reply]
That's weird, but I have to do an additional division by 2 to get the perfect portrait rotation.
`-(detectedFace.rotation.angle.roll * 180 / Math.PI / 2).toFixed(1)`
Why is it working? Could you tell where do those radian values come from? Straight from TF?

```
platform: darwin arm64
agent: NodeJS v16.19.0
tfjs: 4.7.0
```

**Reply by vladmandic** [reply]
face points come from model execution, angles come from matrix math on specific keypoints placed in a plane.
why do you need additional /2? it all depends what do you consider as normalized values.

**Reply by kekoleg** [reply]
I wanted to calculate a face rotation that would horizontally align the eyes, so I use only `angle.roll`. Standard formula without `/ 2` always gives me overrotation.

Well thanks for the reply anyway, I'm curious to try other models and see what happens.

**Reply by kekoleg** [reply]
Oh, now I see why `/ 2` helps – https://github.com/vladmandic/human/commit/5ed2e15a4e27d10d4d0866bb40a9ec4e484b4f3a#diff-cba56667d3c1433909d82a858f5c99730ba61f0e346b9e76c0c6465c2ee846a0
Maybe it's not supposed to be there
